pkg/apis/wmc/v1alpha1/windowsmachineconfig_types.go

```diff
@@ -9,22 +9,37 @@ import (
 
 // WindowsMachineConfigSpec defines the desired state of WindowsMachineConfig
 type WindowsMachineConfigSpec struct {
-	 // Replicas represent how many Windows nodes to be added to the
-        // OpenShift cluster
-        Replicas int `json:"replicas,omitempty"`
-        // InstanceType represents the flavor of instance to be used while
-        // creating the virtual machines. Please note that this is common
-        // across all the Windows nodes in the cluster
-        InstanceType string `json:"instancetype,omitempty"`
-        // sshkeyPair is the sshKeyPair associated with cloudprovider. AWS
-        // asks a keypair to be present for encrypting the Windows VM password
-        sshkeyPair string `json:"sshkeypair"`
-        // sshPrivateKey is the name of the secret which contains ssh key
-        // to decrypt the password
-        sshPrivateKey string `json:"sshprivatekey"`
-        // cloudProviderCreds is the name of the secret which contains
-        // credentials of the cloud provider.
-        cloudProviderCredentials string `json:"cloudprovidercredentials,omitempty"`
+	// Replicas represent how many Windows nodes to be added to the
+	// OpenShift cluster
+	Replicas int `json:"replicas,omitempty"`
+	// InstanceType represents the flavor of instance to be used while
+	// creating the virtual machines. Please note that this is common
+	// across all the Windows nodes in the cluster
+	InstanceType string `json:"instanceType,omitempty"`
+	// AWSProvider holds AWS specific cloud provider information.
+	AWSProvider *AWSProvider `json:"AWSProvider"`
+	// AzureProvider holds Azure specific cloud provider information
+	AzureProvider *AzureProvider `json:"AzureProvider"`
+	// CloudProviderCreds is the name of the secret which contains the
+	// credentials of the cloud provider
+	CloudProviderCredentials string `json:"cloudProviderCredentials,omitempty"`
+}
+
+// AWSProvider holds the information related to AWS cloud provider
+type AWSProvider struct {
+	// SshKeyPair is the sshKeyPair associated with cloudprovider. AWS
+	// asks a keypair to be present for encrypting the Windows VM password
+	SshKeyPair string `json:"sshKeyPair"`
+	// SshPrivateKey is the name of the secret which contains ssh key
+	// to decrypt the password
+	SshPrivateKey string `json:"sshPrivateKey"`
```

Comment:
Same question here.

Reply:
No strong preference. I'll change it to SSH.